src/NHibernate.Test/Async/CacheTest/GetQueryCacheFixture.cs

```diff
@@ -0,0 +1,132 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by AsyncGenerator.
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+using NHibernate.Cache;
+using NHibernate.Cfg;
+using NUnit.Framework;
+using Environment = NHibernate.Cfg.Environment;
+
+namespace NHibernate.Test.CacheTest
+{
+	using System.Threading.Tasks;
+	[TestFixture]
+	public class GetQueryCacheFixtureAsync : TestCase
+	{
+		protected override IList Mappings => new[] { "Simple.hbm.xml" };
+
+		protected override void Configure(Configuration configuration)
+		{
+			configuration.SetProperty(Environment.UseQueryCache, "true");
+			configuration.SetProperty(Environment.CacheProvider, typeof(LockedCacheProvider).AssemblyQualifiedName);
+		}
+
+		[Test]
+		public async Task RetrievedQueryCacheMatchesGloballyStoredOneAsync()
+		{
+			var region = "RetrievedQueryCacheMatchesGloballyStoredOne";
```

Comment:
Initially it was `nameof(RetrievedQueryCacheMatchesGloballyStoredOne`), but this was invalid in the generated async code, due to the `NewType` generation option. Somewhat an async generator bug, but I do not think such a case is worth reporting.